libraries/kotlinandroidlibrary/build.gradle

```diff
@@ -1,8 +1,16 @@
 apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-android-extensions'
 
 dependencies {
   compile deps.external.kotlinStdlib
   testCompile deps.test.junit
   testCompile deps.test.kotlinTest
 }
+
```

Comment:
The logic in this PR will break if we do something like

```
sourceSets.custom.res.srcDirs = ["src/custom_res"]
```

The above is  valid input to the gradle dsl and all it's tasks, but the way we make assumptions about the folder structure in constructing the extension options is incorrect

Reply:
I updated the custom example to use this